src/plugins/ui_feature/autofill.ts

```diff
@@ -238,6 +247,115 @@ export class AutofillPlugin extends UIPlugin {
     }
   }
 
+  private collectBordersData(
+    data: Omit<AutoFillCellCommand, "type">,
+    borders: Record<UID, Border>,
+    bordersPositions: Record<number, Zone[]>
+  ) {
+    const id = getItemId(data.border, borders);
+    if (!(id in bordersPositions)) {
+      bordersPositions[id] = [];
+    }
+    bordersPositions[id].push(positionToZone({ col: data.col, row: data.row }));
+  }
+
+  private collectConditionalFormatsData(
+    sheetId: UID,
+    data: Omit<AutoFillCellCommand, "type">,
+    cfNewRanges: Record<UID, string[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const cfOrigin = this.getters.getRulesByCell(sheetId, originCol, originRow);
+    const xc = toXC(col, row);
+    for (const cf of cfOrigin) {
+      if (!(cf.id in cfNewRanges)) {
+        cfNewRanges[cf.id] = [];
+      }
+      cfNewRanges[cf.id].push(xc);
+    }
+  }
+
+  private collectDataValidationsData(
+    sheetId: UID,
+    data: Omit<AutoFillCellCommand, "type">,
+    dvNewZones: Record<UID, Zone[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const cellPosition = { sheetId, col: originCol, row: originRow };
+    const dvOrigin = this.getters.getValidationRuleForCell(cellPosition);
+    if (!dvOrigin) {
+      return;
+    }
+    if (!(dvOrigin.id in dvNewZones)) {
+      dvNewZones[dvOrigin.id] = [];
+    }
+    dvNewZones[dvOrigin.id].push(positionToZone({ col, row }));
+  }
+
+  private autofillCell(sheetId: UID, data: Omit<AutoFillCellCommand, "type">) {
+    this.dispatch("UPDATE_CELL", {
+      sheetId,
+      col: data.col,
+      row: data.row,
+      content: data.content || "",
+      style: data.style || null,
+      format: data.format || "",
+    });
+    // Still usefull in odoo ATM to autofill field sync
+    this.dispatch("AUTOFILL_CELL", data);
+  }
+
+  private autofillBorders(
+    sheetId: UID,
+    borders: Record<UID, Border>,
+    bordersPositions: Record<number, Zone[]>
+  ) {
+    for (const [borderId, border] of Object.entries(borders)) {
```

Comment:
nitpick: Object.entries isn't great performance-wise (create a new array for each element). Same in the other methods